src/Symfony/Contracts/Service/ServiceProviderInterface.php

```diff
@@ -30,7 +44,7 @@ interface ServiceProviderInterface extends ContainerInterface
      *  * ['foo' => '?'] means the container provides service name "foo" of unspecified type
      *  * ['bar' => '?Bar\Baz'] means the container provides a service "bar" of type Bar\Baz|null
      *
-     * @return string[] The provided service types, keyed by service names
+     * @return array<string, string|class-string<T>> The provided service types, keyed by service names
```

Comment:
`string|class-string<T>` is the same than `string` as `class-string<T>` is a subtype

Reply:
true, but doesn't that still help a but to understand the interface?

Reply:
im not sure i see any practical gain :)

Reply:
ok, removed: for computer it's the same, and for humans there's the comment above already